grails-app/services/com/unifina/service/CommunitySecretService.groovy

```diff
@@ -33,7 +33,12 @@ class CommunitySecretService {
 		if (result == null) {
 			throw new NotFoundException("Community secret not found")
 		}
-		result.name = cmd.name
+		if (cmd.name) {
+			result.name = cmd.name
+		}
+		if (cmd.secret) {
+			result.secret = cmd.secret
```

Comment:
Question: Why do we allow the user to set the secret, instead of always generating it? (also applies to line 19, but can't comment there)

Reply:
it's a matter of order; suppose you'd have to create the secret and deploy it (maybe to an IoT gadget), and only later enable the secret in EE. Not sure if this is a real use case.

Reply:
I'll throw an error then instead. Better than silently 200 and not change the secret IMO.